docs/assessment.md

```diff
@@ -128,6 +128,82 @@ By default, PrairieLearn provides an incentive for students to answer a specific
 
 A question may also set a value to `points` instead of `autoPoints` and `manualPoints`. If this option is used, questions with a `gradingMethod` set to `Manual` will be assigned only manual points, while questions with other grading methods will be assigned only auto points. To avoid ambiguity, it is an error to use both `points` and `autoPoints`, or `points` and `manualPoints`, in the same question. If `points` is used, then `maxPoints` should be used instead of `maxAutoPoints`.
 
+## Question scoring for `Homework` assessments
+
+Assessments with `"type": "Homework"` are designed to allow students to repeatedly practice different variants of questions (that is, different random values in the question) until they achieve mastery. The assumption here is that any student should be able to receive full points for a homework question if they work long enough on it. This section describes in detail how repeated attempts are scored, assuming that the question has `autoPoints`.
+
+The two important question configuration settings are:
+- **`autoPoints`** is the number of points that the student receives for a submission that is fully correct.
+- **`maxAutoPoints`** is the maximum number of points that the student can receive for a question, regardless of how many times they answer it correctly.
+
+The basic idea is that students should receive `autoPoints` each time they solve the question successfully, up to a maximum of `maxAutoPoints`. When partial credit is awarded, a student receives partial points for the question, but only if they are getting closer to a correct answer. For example, if a student repeatedly answers a question with 50% correct answers then they will receive 50% points for the first such answer, but they will not be able to build up points to reach `maxAutoPoints`. They will need to answer the question 100% correctly to be able to advance their points past the 50% mark.
+
+The exact details of how PrairieLearn awards points are as follows. First, a number of variables are computed internally by PrairieLearn as the student answers the question:
+- **Submission percentage score** is the score (between 0% and 100%) that the student receives for any given submission attempt, as evaluated by the question grading specification.
+- **Best current score** is the highest submission percentage score that the student has received since last achieving 100% for a submission. This is initialized to zero.
+- **Question value** is the current maximum number of points that the question is worth. This starts at `autoPoints` and increases by `autoPoints` for each consecutive correct submission. This is initialized to `autoPoints`.
```

Comment:
This is already covered in https://github.com/PrairieLearn/PrairieLearn/blob/f1a2a517b07ecd1bc1c78d52c12c0ec518fa136a/docs/assessment.md?plain=1#L117

I'd suggest merging both for clarity, either here or there. Alternatively, I'd suggest referring to the paragraph there.

Reply:
In 2a6b0b65ab8f28627eaf52dbf629c4a931ddbdb6 I rearranged the earlier section on question points to address this. I split the "question points" section into two sections, one for Homework and one for Exam. I felt that this was clearer than having to always have "for Homework" or "for Exam" scattered throughout the section. It does mean that there is now some duplication, especially regarding manual grading, but I think it's clearer for readers if not for writers.

I'm now treating the "Question points" sections as high-level overviews and the "Question scoring details" sections as all the gory specifics. I cross-reference between them all to hopefully direct people to the right place.

Let me know what you think about this new layout.

I also attempted to address the slack question from @jlumbroso in 6a10b02ce91c2c662d73b6910e13593d4d23c27c -- let me know whether you think I succeeded in this or if there is more we could do there.

Reply:
Personally I think "Question scoring details" could be a sub-section of "Question points" for the specific type, or at least come right after it, to avoid requiring back-and-forth in the description, but I don't feel strongly about this.

Reply:
Yep, I think something like that could be made to work. However, I'm also trying to minimize the level of nesting here, so I'll leave it as is.